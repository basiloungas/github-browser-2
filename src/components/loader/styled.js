import styled, { keyframes } from 'styled-components';

const ldsDualRing = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

export const Loader = styled.div`
  display: inline-block;
  width: 64px;
  height: 64px;

  :after {
    content: " ";
    display: block;
    width: 46px;
    height: 46px;
    margin: 1px;
    border-radius: 50%;
    border: 5px solid black;
    border-color: black transparent black transparent;
    animation: ${ldsDualRing} 1.2s linear infinite;
  }
`;

export const Wrapper = styled.div`
  display: block;
  position: relative;
  width: 100%;
  height: 100%;

  ${Loader} {
    position: absolute;
    top:50%;
    left:50%;
    margin-top: -32px;
    margin-left: -32px;
  }
`
