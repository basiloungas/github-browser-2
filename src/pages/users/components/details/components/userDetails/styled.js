import styled from 'styled-components';
import {Image as BootstrapImage} from 'react-bootstrap';

export const Header = styled.div`
`;

export const Payload = styled.pre`
  margin-top: 30px;
`;

export const ImageWrapper = styled.div`
  width: 120px;
  height: 120px;
  float: left;
  margin-right: 20px;
`;

export const Image = styled(BootstrapImage)`
  object-fit: cover;
`;
