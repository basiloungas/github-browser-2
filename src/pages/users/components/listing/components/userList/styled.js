import styled from 'styled-components';
import {
  Col,
  Button,
  Image as BootstrapImage,
} from 'react-bootstrap';

export const Box = styled(Col)`
  margin-bottom: 30px;
`;

export const ListWrapper = styled.div`
  padding: 10px 0;
  margin: 0 -10px;
`;

export const Image = styled(BootstrapImage)`
  object-fit: cover;
  width: 100%;
  height: 300px; /* optional, but in my case it was good */
`;

export const LoadButton = styled(Button)`
  width: 100%;
  font-size: 1.5em;
  padding: 10px;
`;
