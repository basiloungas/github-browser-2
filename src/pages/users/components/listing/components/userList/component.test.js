import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {ListGroupItem} from 'react-bootstrap';

import Component from './component';
import {LoadButton} from './styled';

describe('pages > Users > Listing > UserList', () => {
  const users = [
    {
      id: 1,
      login: 'login1',
      avatar_url: 'avatar_url1',
    },
    {
      id: 2,
      login: 'login2',
      avatar_url: 'avatar_url2',
    }
  ];

  const defaultProps = {
    users,
    loadMore: () => {},
    isFetchingMore: false,
    history: {push: () => {}},
  }

  test('renders users on the list', () => {
    const wrapper = shallow(<Component {...defaultProps} />);

    expect(wrapper.find(ListGroupItem)).toHaveLength(2);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('renders a "LoadMore" button', () => {
    const wrapper = shallow(<Component {...defaultProps} />);

    expect(wrapper.find(LoadButton)).toHaveLength(1);
  });

  describe('when clicking on a User item', () => {
    test('it calls history.push', () => {
      const pushSpy =  jest.fn();
      const preventDefaultSpy =  jest.fn();
      const event = {
        preventDefault: preventDefaultSpy,
      };
      const props = {
        ...defaultProps,
        history: {
          push: pushSpy
        },
      }

      const wrapper = shallow(<Component {...props} />);
      wrapper
        .find(ListGroupItem)
        .first()
        .simulate('click', event);

      expect(pushSpy.mock.calls).toHaveLength(1);
      expect(preventDefaultSpy.mock.calls).toHaveLength(1);
      expect(pushSpy.mock.calls[0][0]).toEqual('/users/login1');
    });
  });

  describe('when clicking the "LoadMore" button', () => {
    test('it calls the "loadMore" prop', () => {
      const spy =  jest.fn();
      const props = {
        ...defaultProps,
        loadMore: spy
      }

      const wrapper = shallow(<Component {...props} />);
      wrapper
        .find(LoadButton)
        .first()
        .simulate('click');

      expect(spy.mock.calls).toHaveLength(1);
    });
  });

  describe('when "isFetchingMore" is true', () => {
    test('"LoadMore" button is disabled', () => {
      const props = {
        ...defaultProps,
        isFetchingMore: true
      }

      const wrapper = shallow(<Component {...props} />);

      expect(wrapper
        .find(LoadButton)
        .prop('disabled')).toEqual(true);
    });
  });
});
