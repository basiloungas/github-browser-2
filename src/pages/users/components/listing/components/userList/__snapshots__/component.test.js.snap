// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages > Users > Listing > UserList renders users on the list 1`] = `
<React.Fragment>
  <styled.div>
    <Styled(Col)
      key="1"
      lg={3}
      md={4}
      sm={6}
      xs={12}
    >
      <ListGroupItem
        bsClass="list-group-item"
        header="login1"
        href="/users/login1"
        key="1"
        listItem={false}
        onClick={[Function]}
      >
        <Styled(Image)
          alt="login1's thumbnail"
          responsive={true}
          src="avatar_url1"
          thumbnail={true}
        />
      </ListGroupItem>
    </Styled(Col)>
    <Styled(Col)
      key="2"
      lg={3}
      md={4}
      sm={6}
      xs={12}
    >
      <ListGroupItem
        bsClass="list-group-item"
        header="login2"
        href="/users/login2"
        key="2"
        listItem={false}
        onClick={[Function]}
      >
        <Styled(Image)
          alt="login2's thumbnail"
          responsive={true}
          src="avatar_url2"
          thumbnail={true}
        />
      </ListGroupItem>
    </Styled(Col)>
    <Clearfix
      bsClass="clearfix"
      componentClass="div"
    />
  </styled.div>
  <Styled(Button)
    disabled={false}
    onClick={[Function]}
  >
    Load More Users
  </Styled(Button)>
</React.Fragment>
`;
